### PR TITLE
add worker pool switch for each protocol

### PR DESCRIPTION
**Pull request by cch123**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/types/stream.go [hunk 1]
```diff
@@ -180,6 +180,8 @@ type StreamConnection interface {
 	// Protocol on the connection
 	Protocol() api.Protocol
 
+	EnableWorkerPool() bool
```
Comment (nejisama) [inline pkg/types/stream.go:183]: add comment describe it. usually default is true.

File: pkg/stream/xprotocol/conn.go [hunk 2]
```diff
@@ -206,6 +206,11 @@ func (sc *streamConn) Protocol() types.ProtocolName {
 	return protocol.Xprotocol
 }
 
+func(sc *streamConn) EnableWorkerPool() bool {
+	return sc.protocol.EnableWorkerPool()
```
Comment (nejisama) [inline pkg/stream/xprotocol/conn.go:210]: sc.protocol maybe nil

File: pkg/proxy/downstream.go [hunk 3]
```diff
@@ -391,16 +391,25 @@ func (s *downStream) OnReceive(ctx context.Context, headers types.HeaderMap, dat
...
-	// goroutine for proxy
-	if s.proxy.workerpool != nil {
-		// use the worker pool for current proxy
-		// NOTE: should this be configurable?
-		// eg, use config to control Schedule or to ScheduleAuto
-		s.proxy.workerpool.Schedule(task)
-	} else {
-		// use the global shared worker pool
-		pool.ScheduleAuto(task)
+	// s.proxy.serverStreamConn == nil happens when it's in test
```
Comment (taoyuanyuan) [inline pkg/proxy/downstream.go:394]: unused comment

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**